### PR TITLE
Fix Visual Studio build

### PR DESCRIPTION
**Pull request by BenjaminW3**
Should this be added to the 0.3.3 release?

### CODE REVIEW

**Comment by ax3l** [comment]
Yes, please add to the release branch for 0.3.3

**Comment by ax3l** [comment]
@BenjaminW3 there seem to be some other AppVeyor issues. Are these relevant to get 0.3.3 released (aka: is something else broken for MSVC?)

File: include/alpaka/rand/TinyMT/Engine.hpp [hunk 1]
```diff
@@ -28,21 +28,26 @@
 #if BOOST_COMP_CLANG
 #   pragma clang diagnostic push
 #   pragma clang diagnostic ignored "-Wold-style-cast"
-#   pragma clang diagnostic push
```
Comment (ax3l) [inline include/alpaka/rand/TinyMT/Engine.hpp]: that does not work. now it throws again in clang

**Comment by ax3l** [comment]
@BenjaminW3 I will push to this branch to fix

**Comment by ax3l** [comment]
@BenjaminW3 also had to rebase, sorry! Please `git reset --hard HEAD~` your commit locally before fetching my updates.

**Comment by ax3l** [comment]
> still errors for clang, pop each pragma

I reverted your clang change, did not work as planned:

https://travis-ci.org/ComputationalRadiationPhysics/alpaka/jobs/413729129

**Comment by BenjaminW3** [comment]
Thank you for fixing!

File: alpakaConfig.cmake [hunk 2]
```diff
@@ -623,6 +623,7 @@ LIST(APPEND _ALPAKA_LINK_LIBRARIES_PUBLIC "${_ALPAKA_LINK_LIBRARY}")
 
 # Add all the source and include files in all recursive subdirectories and group them accordingly.
 append_recursive_files_add_to_src_group("${_ALPAKA_SUFFIXED_INCLUDE_DIR}" "${_ALPAKA_SUFFIXED_INCLUDE_DIR}" "hpp" _ALPAKA_FILES_HEADER)
+append_recursive_files_add_to_src_group("${_ALPAKA_SUFFIXED_INCLUDE_DIR}" "${_ALPAKA_SUFFIXED_INCLUDE_DIR}" "h" _ALPAKA_FILES_HEADER)
```
Comment (ax3l) [inline alpakaConfig.cmake]: if we do this, we probably have to move the warning suppression into the `.h` header itself...

Comment (ax3l) [reply alpakaConfig.cmake]: We should make Alpaka an installable library now, so we can compile this to an internal object and making the header internal only ;-)

Comment (BenjaminW3) [reply alpakaConfig.cmake]: I do not understand what adding this file to the IDE project has to do with the warnings and the installation. This line should not have any influence on compilation.

Comment (BenjaminW3) [reply alpakaConfig.cmake]: Ah, now I see. The analysis build with the self-contained header check fails. You can remove the line and I will see if I can exclude it from this check later.

**Comment by ax3l** [comment]
> if we do this, we probably have to move the warning suppression into the .h header itself...

updates as described

File: alpakaConfig.cmake (hunk 2 above)
Comment (ax3l) [reply alpakaConfig.cmake]: Seen to late. Do you think the state with moving the suppression into the impl is fine as well?

Comment (ax3l) [reply alpakaConfig.cmake]: Btw: maybe I misinterpreted the first fail on clang and that was the same reason

Comment (BenjaminW3) [reply alpakaConfig.cmake]: I liked the first version with unchanged header more. But both is good. You could try the clang suppression again.

Comment (ax3l) [reply alpakaConfig.cmake]: Me too, but more robust that way now.
Updated regarding clang now as you initially suggested.

**Pull request merged**